source/core/lethe_grid_tools.cc

```diff
@@ -1852,3 +1899,176 @@ LetheGridTools::rotate_mapping(const DoFHandler<3> &dof_handler,
                                const double        &rotation_angle,
                                const Point<3>      &center_of_rotation,
                                const Tensor<1, 3>  &rotation_axis);
+
+
+template <int dim, int spacedim>
+bool
+LetheGridTools::triangle_cells_are_coplanar(
+  const typename Triangulation<dim, spacedim>::active_cell_iterator &cell_A,
+  const typename Triangulation<dim, spacedim>::active_cell_iterator &cell_B,
+  const double                                                       tol)
+{
+  // Point of each triangle cell
+  std::vector<Point<spacedim>> points_A, points_B;
+  points_A.reserve(3);
+  points_B.reserve(3);
+
+  for (unsigned int i = 0; i < 3; ++i)
```

Comment:
This functoin will not work if you cell has spacedim=2 and dim=2. Yet you are instantiating the template for these variants below. If you want to create these variants, I would create empty functions that just AssertThrow(false) with a message that says they are not implemented instead of having something with a false premise of working.